### PR TITLE
drivers: counter: Remove TIME_GECKO_RTCC and doc/finish persistence.

### PR DESCRIPTION
**Pull request by DRuffer**
Remove gecko_rtcc.c driver and references to TIME_GECKO_RTCC. Add CONFIG_PERSISTENT_GECKO_RTCC to turn off persistence by default. Select the SOC_GECKO_RMU with the persistence config not in the soc. Document the low level portions in the counter_gecko_rtcc.c driver. Only disable FSBL on SOCs defined in AN0003, even if not in Zephyr. Add back the settings when persistence is not enabled. Keep date counter command until we upstream, then take it out.

### CODE REVIEW

**Comment by DRuffer** [comment]
This cannot go in until I have also changed all the references to TIME_GECKO_RTCC in all the sources.

**Review by johnlange2 — APPROVED** [review]

File: boards/arm/efm32pg_stk3402a/efm32pg_stk3402a_defconfig [hunk 1]
```diff
@@ -19,3 +19,10 @@ CONFIG_I2C=y
 CONFIG_REBOOT=y
 CONFIG_COUNTER=y
 CONFIG_NOTIFY=y
+CONFIG_LED=y
+CONFIG_LED_PWM=y
+CONFIG_LED_SHELL=y
+CONFIG_PWM=y
+CONFIG_PWM_SHELL=y
+CONFIG_PWM_GECKO=y
+CONFIG_PERSISTENT_GECKO_RTCC=y
```
Comment (jtbaumann) [inline boards/arm/efm32pg_stk3402a/efm32pg_stk3402a_defconfig:28]: Do we really want to add the shells to compile by default with the PG starter kit?

**Comment by DRuffer** [comment]
Replaced with another PR

File: boards/arm/efm32pg_stk3402a/efm32pg_stk3402a_defconfig (hunk 1 above)
Comment (DRuffer-tmo) [reply boards/arm/efm32pg_stk3402a/efm32pg_stk3402a_defconfig:28]: For the testing I am doing now, I'd like to have those configs, but they will likely be removed before I push it upstream

**Pull request closed without merging**